kmcd-prelude.md

```diff
@@ -144,6 +143,134 @@ module KMCD-PRELUDE
          transact ADMIN End . initOut "gold" "Bobby"
          .MCDSteps
       [macro]
+endmodule
+```
 
+```k
+module KMCD-GEN
+    imports KMCD-PRELUDE
+
+    configuration
+      <kmcd-random>
+        <kmcd-properties/>
+        <kmcd-gen>
+          <random> $RANDOMSEED:Int </random>
+          <generator-depth-bound> $GENDEPTH:DepthBound </generator-depth-bound>
+          <generator-next> 0 </generator-next>
+          <generator-current> 0 </generator-current>
+          <generator-remainder> .GenStep </generator-remainder>
+          <generators>
+            <generator multiplicity="*" type="Map">
+              <generator-id> 0 </generator-id>
+              <generator-steps> .GenStep </generator-steps>
+            </generator>
+          </generators>
+          <end-gen> .GenStep </end-gen>
+        </kmcd-gen>
+      </kmcd-random>
+
+    syntax DepthBound ::= Int | "*"
+                        | decrement ( DepthBound ) [function]
+ // ---------------------------------------------------------
+    rule decrement(*) => *
+    rule decrement(N) => N -Int 1
+
+    syntax Int ::= randIntBounded ( Int , Int ) [function]
+ // ------------------------------------------------------
+    rule randIntBounded(RAND, 0)     => 0
+    rule randIntBounded(RAND, BOUND) => RAND modInt BOUND requires BOUND =/=Int 0
+
+    syntax Rat ::= randRat ( Int ) [function]
+ // -----------------------------------------
+    rule randRat(I) => (I modInt 100) /Rat 100
+
+    syntax Rat ::= randRatBounded ( Int , Rat ) [function]
+ // ------------------------------------------------------
+    rule randRatBounded(I, BOUND) => BOUND *Rat randRat(I)
```

Comment:
Is there a reason for using 100 here in `RandRat`? Note that by limiting `randRat` to one hundred possible values here, `randRatBounded` is also limited to a hundred possible values, despite the fact that it's scaled by the bound. My suggestion (Assuming there is no specific reason for picking 100): If the maximum random value that K's `randInt` generates is `RANDMAX`, then define `randRat(I)` as `(I modInt RANDMAX) /Rat (RANDMAX -Int 1). If there is no theoretical maximum, then use any very large number.

Reply:
I picked 100 because it's relatively small and I suspect that most attacks aren't going to be much different based on what the chosen random input is.